Lib/app_interface.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'app'
+class AppInterface
+  ACTIONS = {
+    1 => :list_books,
+    2 => :list_people,
+    3 => :create_person,
+    4 => :create_book,
+    5 => :create_rental,
+    6 => :list_rentals_for_person,
+    7 => :exit_app
+  }.freeze
+  def initialize(app)
+    @app = app
+  end
+
+  def run
+    loop do
+      option = display_menu
+      handle_option(option)
+    end
+  end
+
+  def display_menu
+    puts "\nPlease choose an option:"
+    ACTIONS.each { |key, value| puts "#{key} - #{value.to_s.tr('_', ' ')}" }
+    print 'Option: '
+    gets.chomp.to_i
+  end
+
+  def handle_option(option)
+    action = ACTIONS[option]
+    if action
+      @app.send(action)
+    else
+      puts 'Invalid option. Please try again.'
+    end
+  end
+end
```

Comment:
- You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :100: 
- [ ] In your `app.rb`, in the `create_student and create_teacher ` method, all requests for input and input validation should've been placed in different classes/methods whose responsibilities are to request or validate input. Following the `single responsible rule`, the methods don't need to handle, user input collection, validation, and creating the student or teacher object. I'll suggest you update this. :+1: :100: :wave: